app/controllers/comments_controller.rb

```diff
@@ -6,4 +6,27 @@ def index
   def show
     @comment = Comment.find(params[:id])
   end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new # needed to instantiate the form_for helper in the view
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
```

Comment:
- [ ] I would suggest here to make use of the current user that you have created on application controller, it is a good practice 

```
 @comment = @current_user.comments.new(params)
```